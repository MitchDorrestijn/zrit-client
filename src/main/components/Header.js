import React, { Component } from 'react';
import { Nav, NavItem, NavLink, Row, Col } from 'reactstrap';
import '../css/navbar.css';
import Logo from '../assets/img/logo.gif';

export default class Header extends Component {
  render() {
    return (
      <div>
        <Row className="top-navigation">
          <Col xs="6">
            <figure>
              <img src={Logo} alt="logo Zorgrit" />
            </figure>
          </Col>
          <Col xs="6">
            <span className="float-right logout-btn"><NavLink href="#">Uitloggen</NavLink></span>
          </Col>
        </Row>

        <Nav className="nav-pills nav-justified">
          <NavItem>
            <NavLink href="#">Ritten</NavLink>
          </NavItem>
          <NavItem>
            <NavLink href="#">Chauffeurs</NavLink>
          </NavItem>
          <NavItem>
            <NavLink href="#">Clienten</NavLink>
          </NavItem>
          <NavItem>
            <NavLink href="#">Betalingen</NavLink>
          </NavItem>
          <NavItem>
            <NavLink href="#">Ritten bestellen</NavLink>
          </NavItem>
          <NavItem>
            <NavLink href="#">Over</NavLink>
          </NavItem>
        </Nav>
      </div>
    );
  }
}
