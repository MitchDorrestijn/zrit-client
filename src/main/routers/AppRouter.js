/**
 * React related imports
 */
import React from 'react';
import {
  BrowserRouter,
  Route,
  Switch
} from 'react-router-dom';

/**
 * Self made components imports
 */
import LoginPage from '../container/LoginPage';
import ReadPage from '../container/ReadPage';
import CreatePage from '../container/CreatePage';
import UpdatePage from '../container/UpdatePage';
import ErrorPage from '../container/ErrorPage';
import AboutPage from '../container/AboutPage';

/**
 * Routes that the application knows
 */
const routes = {
  login: '/',
  readZorginstelling: '/read/zorginstelling',
  createZorginstelling: '/create/zorginstelling',
  updateZorginstelling: '/update/zorginstelling',
  error: '/error',
  readChauffeur: '/read/chauffeur',
  createChauffeur: '/create/chauffeur',
  updateChauffeur: '/update/chauffeur',
  readClienten: '/read/client',
  readRideChauffeur: '/read/ride/chauffeur',
  readAllRides: '/read/ride/all',
  aboutPage: '/about',
  readRatings: '/read/ratings',
  createClient: '/create/client',
  updateClient: '/update/client',
  readRideClient: '/read/ride/client',
  createRit: '/create/rit',
  readBetalingen: '/read/betalingen',
  starterpage: '/read'
}

const AppRouter = () => {
  return (
		<BrowserRouter>
	    <Switch>
	      <Route exact path={routes.login} render={(props) => (<LoginPage {...props} routes={routes}/>)} />
        <Route exact path={routes.readZorginstelling} render={(props) => (<ReadPage {...props} routes={routes}/>)} />
        <Route exact path={routes.readChauffeur} render={(props) => (<ReadPage {...props} routes={routes}/>)} />
        <Route exact path={routes.readClienten} render={(props) => (<ReadPage {...props} routes={routes}/>)} />
        <Route exact path={routes.readRatings} render={(props) => (<ReadPage {...props} routes={routes}/>)} />
        <Route exact path={routes.readAllRides} render={(props) => (<ReadPage {...props} routes={routes}/>)} />
        <Route exact path={routes.readRideClient} render={(props) => (<ReadPage {...props} routes={routes}/>)} />
        <Route exact path={routes.readRideChauffeur} render={(props) => (<ReadPage {...props} routes={routes}/>)} />
        <Route exact path={routes.createZorginstelling} render={(props) => (<CreatePage {...props} routes={routes}/>)} />
        <Route exact path={routes.createRit} render={(props) => (<CreatePage {...props} routes={routes}/>)} />
        <Route exact path={routes.createChauffeur} render={(props) => (<CreatePage {...props} routes={routes}/>)} />
        <Route exact path={routes.createClient} render={(props) => (<CreatePage {...props} routes={routes}/>)} />
        <Route exact path={`${routes.updateZorginstelling}/:id`} render={(props) => (<UpdatePage {...props}  routes={routes} />)}/>
        <Route exact path={`${routes.updateClient}/:id`} render={(props) => (<UpdatePage {...props}  routes={routes} />)}/>
        <Route exact path={`${routes.updateChauffeur}/:id`} render={(props) => (<UpdatePage {...props}  routes={routes} />)}/>
        <Route exact path={`${routes.error}`} render={(props) => (<ErrorPage {...props} routes={routes}/>)}/>
        <Route exact path={routes.aboutPage} render={(props) => (<AboutPage {...props} routes={routes}/>)}/>
        <Route exact path={routes.readBetalingen} render={(props) => (<ReadPage {...props} routes={routes}/>)} />
        <Route exact path={routes.starterpage} render={(props) => (<ReadPage {...props} routes={routes}/>)} />
      </Switch>
  	</BrowserRouter>
	);
}

export default AppRouter;
