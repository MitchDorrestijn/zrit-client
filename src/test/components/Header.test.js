//The tests for the Header class are defined here (Header can be found on any page)

import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from '../../main/components/Header';
import AppRouter from '../../main/routers/AppRouter';

const props = {
  readZorginstelling: '/read/zorginstelling'
}

const wrapper = shallow(<Header routes={props} />);

test('<Header /> should match snapshot', () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('<Header /> should render one <Row />', () => {
  expect(wrapper.find('Row').length).toBe(1);
});

test('<Header /> should render two <Col />', () => {
  expect(wrapper.find('Col').length).toBe(2);
});

test('<Header /> should render one figure', () => {
  expect(wrapper.find('figure').length).toBe(1);
});

test('<Header /> should render one image', () => {
  expect(wrapper.find('img').length).toBe(1);
});

test('<Header /> should render <Link /> nine times', () => {
  expect(wrapper.find('Link').length).toBe(9);
});

test('<Header /> should render <Nav /> one time', () => {
  expect(wrapper.find('Nav').length).toBe(1);
});

test('<Header /> should render <NavItem /> seven times', () => {
  expect(wrapper.find('NavItem').length).toBe(7);
});

test('.top-navigation class should be applied to a <Row />', () => {
  expect(wrapper.find('.top-navigation').text()).toBe('<Row />')
});

test('.nav-justified class should be applied to a <Nav />', () => {
  expect(wrapper.find('.nav-justified').text()).toBe('<Nav />')
});

test('.logout-btn class should be applied to a <Link />', () => {
  expect(wrapper.find('.logout-btn').text()).toBe('<Link />')
});
