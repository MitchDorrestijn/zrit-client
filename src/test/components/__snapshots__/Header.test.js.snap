// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div>
  <Row
    className="top-navigation"
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <figure>
        <Link
          replace={false}
          to="/read/zorginstelling"
        >
          <img
            alt="logo Zorgrit"
            src="logo.gif"
          />
        </Link>
      </figure>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <span
        className="float-right logout-btn"
      >
        <Link
          replace={false}
          to="#"
        >
          Uitloggen
        </Link>
      </span>
    </Col>
  </Row>
  <Nav
    className="nav-pills nav-justified"
    tag="ul"
    vertical={false}
  >
    <NavItem
      tag="li"
    >
      <Link
        className="nav-link"
        replace={false}
        to="/read/zorginstelling"
      >
        Zorginstellingen
      </Link>
    </NavItem>
    <NavItem
      tag="li"
    >
      <Link
        className="nav-link"
        replace={false}
        to="#"
      >
        Ritten
      </Link>
    </NavItem>
    <NavItem
      tag="li"
    >
      <Link
        className="nav-link"
        replace={false}
      >
        Chauffeurs
      </Link>
    </NavItem>
    <NavItem
      tag="li"
    >
      <Link
        className="nav-link"
        replace={false}
      >
        Clienten
      </Link>
    </NavItem>
    <NavItem
      tag="li"
    >
      <Link
        className="nav-link"
        replace={false}
        to="#"
      >
        Betalingen
      </Link>
    </NavItem>
    <NavItem
      tag="li"
    >
      <Link
        className="nav-link"
        replace={false}
        to="#"
      >
        Ritten bestellen
      </Link>
    </NavItem>
    <NavItem
      tag="li"
    >
      <Link
        className="nav-link"
        replace={false}
        to="{props.routes.About}"
      >
        Over
      </Link>
    </NavItem>
  </Nav>
</div>
`;
